cluster/log-dump/log-dump.sh

```diff
@@ -68,6 +64,8 @@ function setup() {
     source "${KUBE_ROOT}/cluster/kube-util.sh"
     echo "Detecting project"
     detect-project 2>&1
+  elif [[ "${KUBERNETES_PROVIDER}" == "gke" ]]; then
+    echo "Using 'use_custom_instance_list' with gke, skipping check for LOG_DUMP_SSH_KEY and LOG_DUMP_SSH_USER"
```

Comment:
Er, no. This is wrong. If there's no LOG_DUMP_SSH_KEY, `use_custom_instance_list` can't work. Can we undo this PR instead?

Reply:
Please see below. We don't need the the ssh key in case of gke (even if we're using `use_custom_instances_list`).

Reply:
I'm not going to argue this anymore, but we really shouldn't be using the gcloud assumptions.